src/serial_io.rs

```diff
@@ -1,41 +1,110 @@
-use esp32c3_hal::{
-    interrupt,
-    interrupt::CpuInterrupt,
-    pac::{self, UART0},
-    prelude::*,
-    serial::Instance,
-    Cpu, Serial,
+use heapless::Deque;
+#[cfg(any(target_arch = "riscv32"))]
```

Comment:
we could use critical-section now - no need to worry about the target architecture then